esdt-safe/src/lib.rs

```diff
@@ -349,37 +352,59 @@ pub trait EsdtSafe:
     #[payable("*")]
     #[endpoint(createTransaction)]
     fn create_transaction(
+        &self,
+        to: EthAddress<Self::Api>,
+        opt_min_bridge_amount: OptionalValue<BigUint<Self::Api>>,
+    ) {
+        let transaction_details = self.create_transaction_common(to, opt_min_bridge_amount);
+        let caller = self.blockchain().get_caller();
+
+        self.create_transaction_event(
+            transaction_details.batch_id,
+            transaction_details.tx_nonce,
+            transaction_details.payment_token,
+            transaction_details.actual_bridged_amount,
+            transaction_details.required_fee,
+            caller.as_managed_buffer().clone(),
+            transaction_details.to_address,
+        );
+    }
+
+    /// Create an Ethereum -> MultiversX refund transaction. Only fungible tokens are accepted.
+    ///
+    /// Every transfer will have a part of the tokens subtracted as fees.
+    /// The fee amount depends on the global eth_tx_gas_limit
+    /// and the current GWEI price, respective to the bridged token
+    ///
+    /// fee_amount = price_per_gas_unit * eth_tx_gas_limit
+    #[payable("*")]
+    #[endpoint(createRefundTransaction)]
+    fn create_refund_transaction(
```

Comment:
what happens if the caller has 2 swaps that will need to be refunded? Do we have a test for this?